S21/rebelDi/tests/diana/WordToLowerCaseTest.java

```diff
@@ -0,0 +1,72 @@
+package diana;
+
+import org.junit.Test;
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.StringTokenizer;
+import static org.junit.Assert.*;
+
+public class WordToLowerCaseTest {
+
+    /**
+     * This is the test method for WordToLowerCase class,
+     * that tests the result of changeWordToLowerCase()
+     */
+    @Test
+    public void changeWordToLowerCase() throws IOException {
+        List<String[]> wordsToProcess = openTSV();
+        for(String[] wordData:wordsToProcess){
+            String wordForLowerCase = wordData[0];
+            String language = wordData[1];
+            String correctWordInLowerCase = wordData[2];
+
+            System.out.println("Word for lowercase: " + wordForLowerCase);
+
+            String wordInLowerCase = WordToLowerCase.changeWordToLowerCase(wordForLowerCase, language);
+            assertEquals("The word was processed correctly", correctWordInLowerCase, wordInLowerCase);
+        }
+    }
+
+    /**
+     * This is the function for opening the tsv file with examples of words to convert
+     *
+     * @return all the data from file
+     */
+    public static List<String[]> openTSV () throws IOException {
+        List<String[]> listOfDataFromFile = new ArrayList<>();
+
+        StringTokenizer stringTokenizer; // used to get data from row for each column
+        BufferedReader tsvFile = new BufferedReader(new FileReader
+                (System.getProperty("user.dir") + "\\tests\\resources\\tests.tsv"));
+        String dataRow = tsvFile.readLine(); // Read first line
+
+        int iteration = 1;
+        String[] row = new String[3]; // used to get one line of data to the list
+        while (dataRow != null){
```

Comment:
Multiple loops in a single function, kind of hard for me to read each condition.

Reply:
Do you think if I add a comment on how each loop works it would be easier to understand? I just don't think it's a good idea to separate the loops